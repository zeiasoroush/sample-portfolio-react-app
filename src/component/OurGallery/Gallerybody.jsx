import React, { Component } from 'react';
import OwlCarousel from 'react-owl-carousel2';
import 'react-bootstrap';
import './Owl.css';

class Gallerybody extends Component {
  render() {
    const options = {
      margin: 15,

      responsive: {
        0: {
          items: 1
        },
        600: {
          items: 3
        },
        1000: {
          items: 4
        }
      }
    };
    return (
      <OwlCarousel
        className="owl-carousel owl-theme"
        pulldrag="true"
        options={options}
      >
        {this.props.contentBody.map((content, index) => {
          return (
            <div key={index} className="col-md-12">
              <div className="featured-item">
                <a href="https://www.google.com">
                  <img src={content.src} alt={content.alt} />
                </a>
                <div className="down-content">
                  <h4>{content.h4}</h4>
                  <h6>{content.h6}</h6>
                </div>
              </div>
            </div>
          );
        })}
      </OwlCarousel>
    );
  }
}
export default Gallerybody;
