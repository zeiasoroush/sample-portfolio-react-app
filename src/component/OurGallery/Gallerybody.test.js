import React from 'react';
import { shallow, mount } from 'enzyme';
import Gallerybody from './Gallerybody';

describe('it should render gallery page content', () => {
  let component, find, props;
  beforeEach(() => {
    props = {
      gallerycontent: [
        {
          id: 1,
          src: require('../../assets/images/item-01.jpg'),
          alt: 'item1',
          h4: 'Donec non sagittis',
          h6: '$25.00'
        },
        {
          id: 2,
          src: require('../../assets/images/item-02.jpg'),
          alt: 'item2',
          h4: 'Nulla a pharetra',
          h6: '$35.00'
        },
        {
          id: 3,
          src: require('../../assets/images/item-03.jpg'),
          alt: 'item3',
          h4: 'Aliquam convallis',
          h6: '$45.00'
        },
        {
          id: 4,
          src: require('../../assets/images/item-04.jpg'),
          alt: 'item4',
          h4: 'Vivamus vittae #4',
          h6: '$55.00'
        },
        {
          id: 5,
          src: require('../../assets/images/item-05.jpg'),
          alt: 'item5',
          h4: 'Vivamus vittae #6',
          h6: '$65.00'
        },
        {
          id: 6,
          src: require('../../assets/images/item-06.jpg'),
          alt: 'item6',
          h4: 'Vivamus vittae #8',
          h6: '$75.00'
        }
      ]
    };
    component = shallow(<Gallerybody contentBody={props.gallerycontent} />);
  });
  it('should render content', () => {
    expect(component).toMatchSnapshot();
    find = component.find('a');
    expect(find).toHaveLength(6);
  });
});
