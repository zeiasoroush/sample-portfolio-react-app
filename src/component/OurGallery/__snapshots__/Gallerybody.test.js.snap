// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render gallery page content should render content 1`] = `ShallowWrapper {}`;
